src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -456,5 +456,47 @@ public static int Count<T>(this ImmutableArray<T> items, Func<T, bool> predicate
 
             return count;
         }
+
+        internal static Dictionary<K, ImmutableArray<T>> ToDictionary<K, T>(this ImmutableArray<T> items, Func<T, K> keySelector, IEqualityComparer<K> comparer = null)
+        {
+            if (items.Length == 1)
```

Comment:
I just copied the implementation from the builder. Maybe that case was not common.
I will check if it is common enough. Perhaps we need to add special case to the builder too.